### PR TITLE
[CFD-72] Day Schedule

### PR DESCRIPTION
**Pull request by helenaglow**
### Description
Made a schedule component based for activities!
- [X] size of activity components changes based on duration
   >  essentially I calculate the size respective to the difference in time line so it fits as it would in a calendar
- [X] schedule and end hours are dependent on the activities (default 9 am start and 5 pm end)
   >  if there is an event earlier or later than the default, the times will adjust to fit them
- [X] aligns the activity components on the schedule based on start time

### Reason for Change
Completes CFD-72

### Type of change
- [X] New feature (non-breaking change which adds functionality)

### Testing
Tested using `index.tsx`.

In imports: 
```
import type { RouterOutputs } from "~/utils/api";
import Schedule from "./schedule";
```
Add these constants/types:
```
type GetActivityOutput = RouterOutputs["activity"]["getSchedule"][number];

const myActivityProps: GetActivityOutput[] = [
    {
      id: 1,
      name: "Meeting",
      durationMinutes: 60,
      location: "Conference Room",
      day: new Date(),
      startTime: new Date("2023-10-09T08:00:15.000"),
      leader: "me",
    },
    {
      id: 2,
      name: "Meeting 2",
      durationMinutes: 180,
      location: "Conference Room",
      day: new Date(),
      startTime: new Date("2023-10-09T10:30:15.000"),
      leader: "me",
    },
  ];
```

In `index` return add:
```
<View className="h-full w-full p-4">
        <Schedule activities={myActivityProps} />
</View>
```
Result: 

⚠️ I experienced a lot of issues with expo not picking up my tailwind css code. If you wanna try out the code and do not get expected result, let me know!

### CODE REVIEW

**Comment by helenaglow** [comment]
This code is dependent on the activities PR.

Also, I am still trying to fix that straight edge on the activity components on the right without hardcoding a width. So besides that...

**Comment by BakerWJ** [comment]
The component looks good -- don't worry about getting styles perfect, as we'll be updating the styles once we get the hi-fis

**Comment by helenaglow** [comment]
Fixed schedule to work with updated activity

**Comment by helenaglow** [comment]
Done ✅

File: apps/expo/src/app/schedule.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+const MAX_TIME = 18;
+const MIN_TIME = 9;
+const LINE_HEIGHT = 12;
+
+// Activity type based on query of get schedule
+type GetActivityOutput = RouterOutputs["activity"]["getSchedule"][number];
+
+//
+interface ScheduleProps {
+  activities: GetActivityOutput[];
```
[the remark this answers is not in the document]
Comment (helenaglow) [reply apps/expo/src/app/schedule.tsx:18]: Ok makes sense

File: apps/expo/src/app/schedule.tsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+import type { RouterOutputs } from "~/utils/api";
+import Activity from "./activity";
+
+// GLOBAL CONSTANTS
+const MAX_TIME = 18;
+const MIN_TIME = 9;
+const LINE_HEIGHT = 12;
+
+// Activity type based on query of get schedule
+type GetActivityOutput = RouterOutputs["activity"]["getSchedule"][number];
```
[the remark this answers is not in the document]
Comment (helenaglow) [reply apps/expo/src/app/schedule.tsx]: Good call

**Comment by helenaglow** [comment]
Hi @BakerWJ i updated the PR last week, do you mind looking at it when you have time?

**Review by BakerWJ — DISMISSED** [review]
looks good - there are some merge conflicts from dependencies updating, which you can fix by rebasing + rerunning pnpm i

**Review by BakerWJ — APPROVED** [review]

**Pull request merged**